src/include/btree.i

```diff
@@ -1083,6 +1083,13 @@ __wt_page_can_evict(WT_SESSION_IMPL *session,
 	/* Pages that have never been modified can always be evicted. */
 	if (mod == NULL)
 		return (true);
+	 /*
+	  * If the page is clean but has modifications that appear too
+	  * new to evict, skip it.
+	  */
+	if (!__wt_page_is_modified(page) &&
+	    !__wt_txn_visible_all(session, mod->rec_max_txn))
+	        return (false);
```

Comment:
If you change this to `if (!__wt_txn_visible_all(session, mod->rec_max_txn)) ...`, is that sufficient to avoid the assertion (without the unrelated change to `evict_lru.c`)?